src/kernel/src/fs/null/mod.rs

```diff
@@ -0,0 +1,126 @@
+use super::{FsConfig, FsOps, LookupError, Mount, MountFlags, MountOpts, VPathBuf, Vnode};
+use crate::errno::{Errno, EDEADLK, EINVAL, EOPNOTSUPP};
+use crate::fs::VPath;
+use crate::ucred::Ucred;
+use bitflags::bitflags;
+use std::num::NonZeroI32;
+use std::sync::Arc;
+use thiserror::Error;
+
+mod vnode;
+
+pub fn mount(
+    _conf: &'static FsConfig,
+    _cred: &Arc<Ucred>,
+    _path: VPathBuf,
+    _parent: Option<Arc<Vnode>>,
+    mut opts: MountOpts,
+    flags: MountFlags,
+) -> Result<Mount, Box<dyn Errno>> {
+    if flags.intersects(MountFlags::MNT_ROOTFS) {
+        Err(MountError::RootFs)?;
+    }
+
+    if flags.intersects(MountFlags::MNT_UPDATE) {
+        if opts.remove("export").is_some_and(|opt| opt.unwrap()) {
+            todo!("null_mount with MNT_UPDATE and export = true")
+        } else {
+            Err(MountError::NoExport)?
+        }
+    }
+
+    let target: Box<str> = opts
+        .remove("target")
+        .or_else(|| opts.remove("from"))
+        .ok_or(MountError::NoTarget)?
+        .unwrap();
+
+    if target.is_empty() {
+        Err(MountError::EmptyTarget)?;
+    }
+
+    let _target: &VPath = target.as_ref().try_into().unwrap();
```

Comment:
If we accept `target` and `from` as a `VPathBuf` we can eliminate this. We also don't need the above statement that check if target is empty.